instrumentation/opentelemetry/init.go

```diff
@@ -141,15 +148,17 @@ func makeExporterFactory(cfg *config.AgentConfig) func() (sdktrace.SpanExporter,
 		}
 
 		return func() (sdktrace.SpanExporter, error) {
-			return zipkin.New(
+			globalExporter, globalExporterErr = zipkin.New(
 				cfg.GetReporting().GetEndpoint().GetValue(),
 				zipkin.WithClient(client),
 			)
+			return globalExporter, globalExporterErr
 		}
 	case config.TraceReporterType_LOGGING:
 		return func() (sdktrace.SpanExporter, error) {
 			// TODO: Define if endpoint could be a filepath to write into a file.
-			return stdouttrace.New(stdouttrace.WithPrettyPrint())
+			globalExporter, globalExporterErr = stdouttrace.New(stdouttrace.WithPrettyPrint())
```

Comment:
Does this need to be thread safe? If yes, do we need to worry about races resulting in some dangling exporters or that’s not a big deal?

Reply:
Maybe not since it would set up in the Init method. Unless all the consumers do not call Init. But I will shelf this one for now since I did not observe any remarkable improvement with one exporter vs multiple exporters.

Reply:
Not opposed to shelving it, but why no improvement? This should help to reduce number of export connetions?

Reply:
I didn't have time to verify that using lsof or some other networking tool. I suspect reducing clients does not mean reducing connections as well. The clients maybe fewer but the connection count may still be the same. I would want to verify it next.